packages/opentelemetry-sdk-trace-base/src/Sampler.ts

```diff
@@ -53,6 +59,13 @@ export interface SamplingResult {
    * can safely cache the returned value.
    */
   attributes?: Readonly<SpanAttributes>;
+  /**
+   * A {@link TraceState} that will be associated with the {@link Span} through
+   * the new {@link SpanContext}. Samplers SHOULD return the passed-in
+   * TraceState if they do not intend to change it. Leaving the value undefined
```

Comment:
Question: The "passed-in TraceState", is it assumed that the sampler will read it off of the context that is passed in? Would it make sense to pass the tracestate as a separate argument to `shouldSample`?

Reply:
Alternatively, perhaps this comment should mention that the TraceState from the passed-in context, just to avoid confusion?

Reply:
I think it would be quite confusing to pass `context` and some information which is usually on context as an additional parameter. Samplers have to decide then which source they prefer and if they should use the other as fallback,

Reply:
I agree, but reading this comment made me wonder, "passed-in where?" and immediately assumed that the function signature is out of date with this comment. I agree that having a single source of truth is preferred, so perhaps the comment should be modified to clarify what "passed-in" refers to?

Reply:
Good point, updated the doc comment to clarify